### ISSUE TITLE
spanner: calling db.Single() causes panic

### ISSUE THREAD

**Issue by majelbstoat** [googleapis/google-cloud-go#1944]
**Client**

Spanner

**Environment**

Alpine Docker on local k8s

**Code**

e.g.
```go
package main

func main() {
  ctx := context.Background()
  conn := fmt.Sprintf("projects/xxxxxx/instances/xxxxxxx/databases/xxxxxxxx")
  config := spanner.SessionPoolConfig{MinOpened: 5, WriteSessions: float64(1)}
  db, err := spanner.NewClientWithConfig(ctx, conn, spanner.ClientConfig{SessionPoolConfig: config})
  if err != nil {
    logger.WithError(err).Fatal("Could not create a Spanner client.")
  }
  defer db.Close()

  single := db.Single()

  // ... do something with single
}
```

**Expected behavior**

I get a single use readonly transaction

**Actual behavior**

panic:

```
runtime error: invalid memory address or nil pointer dereference:
 	/usr/local/Cellar/go/1.12.7/libexec/src/runtime/panic.go:522 +0x1b5
 cloud.google.com/go/spanner.(*Client).Single(0x0, 0x7fa0f5d61460)
 	/usr/local/lib/go/src/cloud.google.com/go/spanner/client.go:324 +0x3f
```

**Additional context**

The offending line is: https://github.com/googleapis/google-cloud-go/blob/master/spanner/client.go#L324

Happening with latest master. This also happened with a previous version of the client, on the equivalent line assigning `sp` to `c.idleSessions`

It only appears to be `Single()` that is affected. I can successfully call:

```go
  txn := s.db.ReadOnlyTransaction()
  defer txn.Close()

  // ... ok
```

Thinking it might be the session pool config, I also tried with:

```go
  db, err := spanner.NewClient(ctx, conn)
```

But it gives the same panic in the same place.

**Comment by hengfengli** [comment]
Hi @majelbstoat, I tested with the following code against the latest master and it works fine for me.

```go
package main

import (
	"context"
	"fmt"
	"log"

	"cloud.google.com/go/spanner"
)

func main() {
	ctx := context.Background()
	conn := "projects/xxxx/instances/xxxx/databases/test-db"

	config := spanner.SessionPoolConfig{MinOpened: 5, WriteSessions: float64(1)}
	db, err := spanner.NewClientWithConfig(ctx, conn, spanner.ClientConfig{SessionPoolConfig: config})
	if err != nil {
		log.Fatal("Could not create a Spanner client.")
	}
	defer db.Close()

	row, err := db.Single().ReadRow(ctx, "Users",
		spanner.Key{"test"}, []string{"Name", "MySkills", "NeededSkills", "OfficeLocations", "PreferredDays"})
	if err != nil {
		log.Fatal(err)
	}

	var name string
	var mySkills, neededSkills, officeLocations []string
	var preferredDays string

	if err := row.Columns(&name, &mySkills, &neededSkills, &officeLocations, &preferredDays); err != nil {
		log.Fatalf("Unpacking string value: %v", err)
	}

	fmt.Println(name)
	fmt.Println(mySkills)
	fmt.Println(neededSkills)
	fmt.Println(officeLocations)
	fmt.Println(preferredDays)
}
```

I can only think of two things at the moment: 
* Do you have any idea which version of cloud spanner client library are you using? You probably can find it in `go.mod` or you can use `go mod` to specify a specific version of cloud spanner. 
* Can you try to test your code on your local machine to see if it works?

**Comment by majelbstoat** [comment]
uh, so turns out this is PEBKAC, and it turns out what I thought was a valid DB client was actually a nil pointer 😬(My example above was a stripped down version that omitted some complexity I thought wasn't important.)

I really would have expected go to panic on the line that has:

```
  single := db.Single()
```

but it only actually panics inside the method 🤷🏼‍♂️ It makes sense now that I understand what's happening.

tl;dr go's typed nil pointers are confusing, and I'm sorry for the noise.
